homeassistant/components/ezviz/select.py

```diff
@@ -46,45 +138,43 @@ async def async_setup_entry(
     ]
 
     async_add_entities(
-        EzvizSelect(coordinator, camera)
+        EzvizSelect(coordinator, camera, ALARM_SOUND_MODE_SELECT_TYPE)
         for camera in coordinator.data
         for switch in coordinator.data[camera]["switches"]
-        if switch == SELECT_TYPE.supported_switch
+        if switch == ALARM_SOUND_MODE_SELECT_TYPE.supported_switch
+    )
+
+    async_add_entities(
+        EzvizSelect(coordinator, camera, BATTERY_WORK_MODE_SELECT_TYPE)
+        for camera in coordinator.data
+        if coordinator.data[camera]["device_category"]
+        == DeviceCatagories.BATTERY_CAMERA_DEVICE_CATEGORY.value
     )
 
 
 class EzvizSelect(EzvizEntity, SelectEntity):
     """Representation of a EZVIZ select entity."""
 
+    entity_description: EzvizSelectEntityDescription
+
     def __init__(
         self,
         coordinator: EzvizDataUpdateCoordinator,
         serial: str,
+        description: EzvizSelectEntityDescription,
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator, serial)
-        self._attr_unique_id = f"{serial}_{SELECT_TYPE.key}"
-        self.entity_description = SELECT_TYPE
+        self._attr_unique_id = f"{serial}_{description.key}"
+        self.entity_description = description
 
     @property
     def current_option(self) -> str | None:
         """Return the selected entity option to represent the entity state."""
-        sound_mode_value = getattr(
-            SoundMode, self.data[self.entity_description.key]
-        ).value
-        if sound_mode_value in [0, 1, 2]:
-            return self.options[sound_mode_value]
-
-        return None
+        return self.entity_description.action_handler.current_option(self)
 
     def select_option(self, option: str) -> None:
         """Change the selected option."""
-        sound_mode_value = self.options.index(option)
-
-        try:
-            self.coordinator.ezviz_client.alarm_sound(self._serial, sound_mode_value, 1)
-
-        except (HTTPError, PyEzvizError) as err:
-            raise HomeAssistantError(
-                f"Cannot set Warning sound level for {self.entity_id}"
-            ) from err
+        return self.entity_description.action_handler.select_option(
+            self, self._serial, option
+        )
```

Comment:
Honestly, keep the try catch here. It will help you remove the try catch from the function you are executing. You can use the `key` of the entity description to create a translatable error.